### ISSUE TITLE
Error when trying to use pslam.phar

### ISSUE THREAD

**Issue by ADmad** [vimeo/psalm#600]
- Downloaded `pslam.phar` in a directory which is in my environment path.
- Ran `pslam.phar --init` in the project direcotry which went fine.
- Ran `pslam.phar src/` which resulted in below error:
```
$ psalm.phar src/
Scanning files...
Analyzing files...
PHP Fatal error:  Uncaught InvalidArgumentException: Could not get class storage for cake\database\type&cake\database\type\expressiontypeinterface in phar:///home/admad/bin/psalm.phar/src/Psalm/Provider/ClassLikeStorageProvider.php:31
Stack trace:
#0 phar:///home/admad/bin/psalm.phar/src/Psalm/Codebase/Methods.php(52): PsalmPhar\Psalm\Provider\ClassLikeStorageProvider->get('cake\\database\\t...')
#1 phar:///home/admad/bin/psalm.phar/src/Psalm/Codebase.php(462): PsalmPhar\Psalm\Codebase\Methods->methodExists('cake\\database\\t...', NULL)
#2 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(282): PsalmPhar\Psalm\Codebase->methodExists('cake\\database\\t...')
#3 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(128): PsalmPhar\Psalm\Checker\FunctionChecker::getArrayMapReturnType(Object(PsalmPhar\Psalm\Checker\StatementsChecker), Array, Object(PsalmPhar\Psalm\CodeLocation), Array)
#4 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(67): PsalmPhar\Psalm\Checker\Func in phar:///home/admad/bin/psalm.phar/src/Psalm/Provider/ClassLikeStorageProvider.php on line 31

Fatal error: Uncaught InvalidArgumentException: Could not get class storage for cake\database\type&cake\database\type\expressiontypeinterface in phar:///home/admad/bin/psalm.phar/src/Psalm/Provider/ClassLikeStorageProvider.php:31
Stack trace:
#0 phar:///home/admad/bin/psalm.phar/src/Psalm/Codebase/Methods.php(52): PsalmPhar\Psalm\Provider\ClassLikeStorageProvider->get('cake\\database\\t...')
#1 phar:///home/admad/bin/psalm.phar/src/Psalm/Codebase.php(462): PsalmPhar\Psalm\Codebase\Methods->methodExists('cake\\database\\t...', NULL)
#2 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(282): PsalmPhar\Psalm\Codebase->methodExists('cake\\database\\t...')
#3 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(128): PsalmPhar\Psalm\Checker\FunctionChecker::getArrayMapReturnType(Object(PsalmPhar\Psalm\Checker\StatementsChecker), Array, Object(PsalmPhar\Psalm\CodeLocation), Array)
#4 phar:///home/admad/bin/psalm.phar/src/Psalm/Checker/FunctionChecker.php(67): PsalmPhar\Psalm\Checker\Func in phar:///home/admad/bin/psalm.phar/src/Psalm/Provider/ClassLikeStorageProvider.php on line 31
```

The project/codebase I tried to run is `3.next` branch of [cakephp/cakephp](https://github.com/cakephp/cakephp).

**Comment by muglug** [comment]
This is related to intersection type support – I've added

**Comment by muglug** [comment]
Sorry about that - a `__toString` was getting invoked inappropriately. I'll create a new release and phar tomorrow with the fix included. In the meantime you can either build the phar yourself or use `composer require --dev vimeo/psalm:dev-master`

**Comment by ADmad** [comment]
@muglug Thanks for the quick fix.

Regarding the phar file, trying to use it the first time I got error `..failed to open stream: bz2 extension is required for bzip2 compressed .phar file..`, which of course was fixed by installing the bz2 extension for PHP.

But I never encountered such issue when using phars of other well known projects like phpunit or composer, so wouldn't it be better to generate your phar in similar manner as the other popular PHP projects do?
